### ISSUE TITLE
operands are out of order

### ISSUE THREAD

**Issue by ThinkOpenly** [ThinkOpenly/sail#26]
I made an unfortunate choice long ago to pull operands from the "function signature" of the `function clause execute` instances. The order of the inputs there need not match the order of the operands in the assembly syntax, and indeed there might be inputs which are not operands. For example, the `vaadd.vv` instruction's assembly syntax is:
```
vaadd.vv vd,vs2,vs1[,vm]
```

However, the `function clause execute` signature looks like:
```
function clause execute(MVVTYPE(funct6, vm, vs2, vs1, vd))
```
Here, there is an extra input, "funct6", and the operands are in fairly random order. This is also true of the signature for the other stanzas, including `mapping clause encdec` and `mapping clause assembly`.

The operands, not surprisingly, are well represented in the actual assembly syntax in `mapping clause assembly`:
```
mapping clause assembly = MVVTYPE(funct6, vm, vs2, vs1, vd)
  <-> mvvtype_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ sep() ^ vreg_name(vs1) ^ maybe_vmask(vm)
```
The latter line there translates fairly easily into `vaadd.vv vd,vs2,vs1[,vm]`.

I think we need to move the operand extraction from the `function clause execute` processing to the `mapping clause assembly` processing, if possible.

**Comment by Shubhf** [comment]
Hey Can I take up this issue?

**Comment by ThinkOpenly** [comment]
> Hey Can I take up this issue?

@Shubhf Thanks for the offer. @Linda-Njau has been working on this, and it quite close, in #34 .

**Comment by ThinkOpenly** [comment]
Fixed by #34, that is now merged.

Closing...
